server/src/test/java/io/druid/curator/announcement/AnnouncerTest.java

```diff
@@ -60,6 +60,7 @@ public void tearDown()
   public void testSanity() throws Exception
   {
     curator.start();
+    curator.blockUntilConnected();
```

Comment:
since this is unbounded blocking, can you add a timeout to this (and other) tests with @Test ?